sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRegistryBlobClientLiveTests.cs

```diff
@@ -0,0 +1,184 @@
+            var tags = artifact.GetTagPropertiesCollectionAsync();
+            var count = await tags.CountAsync();
+            Assert.AreEqual(1, count);
+            var firstTag = await tags.FirstAsync();
+            Assert.AreEqual(tag, firstTag.Name);
+
+            // Clean up
+            await client.DeleteManifestAsync(digest);
+        }
+
+        private async Task UploadManifestPrerequisites(ContainerRegistryBlobClient client)
+        {
+            var layer = "654b93f61054e4ce90ed203bb8d556a6200d5f906cf3eca0620738d6dc18cbed";
+            var config = "config.json";
+            var basePath = Path.Combine(TestContext.CurrentContext.TestDirectory, "Data", "oci-artifact");
+
+            // Upload config
+            using (var fs = File.OpenRead(Path.Combine(basePath, config)))
+            {
+                var uploadResult = await client.UploadBlobAsync(fs);
+            }
+
+            // Upload layer
+            using (var fs = File.OpenRead(Path.Combine(basePath, layer)))
+            {
+                var uploadResult = await client.UploadBlobAsync(fs);
+            }
+        }
+
+        private static void ValidateManifest(OciManifest manifest)
+        {
+            // These are from the values in the Data\oci-artifact\manifest.json file.
+            Assert.IsNotNull(manifest);
+
+            Assert.IsNotNull(manifest.Config);
+            Assert.AreEqual("application/vnd.acme.rocket.config", manifest.Config.MediaType);
+            Assert.AreEqual("sha256:d25b42d3dbad5361ed2d909624d899e7254a822c9a632b582ebd3a44f9b0dbc8", manifest.Config.Digest);
+            Assert.AreEqual(171, manifest.Config.Size);
+
+            Assert.IsNotNull(manifest.Layers);
+            Assert.AreEqual(1, manifest.Layers.Count);
+            Assert.AreEqual("application/vnd.oci.image.layer.v1.tar", manifest.Layers[0].MediaType);
+            Assert.AreEqual("sha256:654b93f61054e4ce90ed203bb8d556a6200d5f906cf3eca0620738d6dc18cbed", manifest.Layers[0].Digest);
+            Assert.AreEqual(28, manifest.Layers[0].Size);
+        }
+
+        [RecordedTest, NonParallelizable]
+        public async Task CanUploadBlob()
+        {
+            // Arrange
+            var client = CreateBlobClient("oci-artifact");
+            var blob = "654b93f61054e4ce90ed203bb8d556a6200d5f906cf3eca0620738d6dc18cbed";
+            var path = Path.Combine(TestContext.CurrentContext.TestDirectory, "Data", "oci-artifact", blob);
+
+            string digest = default;
+            // Act
+            using (var fs = File.OpenRead(path))
+            {
+                var uploadResult = await client.UploadBlobAsync(fs);
+                digest = uploadResult.Value.Digest;
+            }
+
+            // Assert
+            var downloadResult = await client.DownloadBlobAsync(digest);
+            Assert.AreEqual(digest, downloadResult.Value.Digest);
+            Assert.AreEqual(28, downloadResult.Value.Content.Length);
```

Comment:
Where does `28` come from?

Reply:
In some places, it's hard-coded in the manifest file.  I've updated others to get it from the length of the stream where we have it.